modelconverter/utils/onnx_tools.py

```diff
@@ -176,3 +181,975 @@ def onnx_attach_normalization_to_inputs(
+                    cast_node = gs.Node(
+                        op="Cast",
+                        attrs={"to": new_dtype},
+                        inputs=[first_input],
+                        outputs=[gs.Variable(name=f"{node.name}/Cast_output")],
+                        name=f"{node.name}/Cast",
+                    )
+                    new_nodes.append(cast_node)
+                    first_input = cast_node.outputs[0]
+
+                new_nodes.append(
+                    gs.Node(
+                        op="Mul",
+                        inputs=[
+                            first_input,
+                            gs.Constant(
+                                name=f"{second_input.name}/Subtitute",
+                                values=np.array(new_cost_val, dtype=new_dtype),
+                            ),
+                        ],
+                        outputs=[gs.Variable(name=f"{node.name}/Mul_output")],
+                        name=f"{node.name}/To_Mul",
+                    )
+                )
+
+            return new_nodes
+
+        nodes_to_add = []
+        nodes_to_remove = []
+        connections_to_fix = []
+
+        for node in self.onnx_gs.nodes:
+            if node.op == source_node:
+                constant = self.get_constant_value(node, constant_map)
+                if constant is not None:
+                    _, const_idx = constant
+                    new_nodes = create_new_node(node, target_node, const_idx)
+                    if new_nodes is not None:
+                        if len(new_nodes) > 1:
+                            nodes_to_add.extend(new_nodes)
+                            connections_to_fix.append(
+                                (
+                                    node.outputs[0],
+                                    new_nodes[1].outputs[0],
+                                )
+                            )
+                        else:
+                            nodes_to_add.append(new_nodes[0])
+                            connections_to_fix.append(
+                                (
+                                    node.outputs[0],
+                                    new_nodes[0].outputs[0],
+                                )
+                            )
+                        nodes_to_remove.append(node)
+
+        self.graph_cleanup(nodes_to_add, nodes_to_remove, connections_to_fix)
+        self.onnx_model = gs.export_onnx(self.onnx_gs)
+
+        self.optimize_onnx(passes=["fuse_add_bias_into_conv"])
+
+    def fuse_add_mul_to_bn(self) -> None:
+        """Fuse Add/Sub and Mul nodes that come immediately after a Conv node into a
+        BatchNormalization node.
+
+        The fusion patterns considered are:
+        1. Conv -> Add -> Mul
+        2. Conv -> Sub -> Mul
+        3. Conv -> Mul -> Add
+        4. Conv -> Mul -> Sub
+        5. Conv -> Mul
+        6. Conv -> Add
+        7. Conv -> Sub
+        """
+
+        FUSION_PATTERNS = [
```

Comment:
I would only keep the following patterns: 
(Conv, Add, Mul)
(Conv, Mul, Add)
(Conv, Mul)
(Conv, Add)

There should be no sub or div layers when running this function due to the replace function that is ran before. This would make it easier for debugging later and there are less options to take care of in this file. 

Potentially I would even move (Conv, add) and (Conv, Mul) to `fuse_single_add_mul_to_conv` as there we also fuse single layers instead of making bn layers out of these.